Sources/WalletConnectRelay/RelayClient.swift

```diff
@@ -98,65 +110,69 @@ public final class RelayClient {
 
     /// Completes when networking client sends a request, error if it fails on client side
     public func publish(topic: String, payload: String, tag: Int, prompt: Bool = false) async throws {
-        let params = RelayJSONRPC.PublishParams(topic: topic, message: payload, ttl: defaultTtl, prompt: prompt, tag: tag)
-        let request = JSONRPCRequest<RelayJSONRPC.PublishParams>(method: RelayJSONRPC.Method.publish.method, params: params)
-        logger.debug("Publishing Payload on Topic: \(topic)")
-        let requestJson = try request.json()
-        try await dispatcher.send(requestJson)
+        let request = Publish(params: .init(topic: topic, message: payload, ttl: defaultTtl, prompt: prompt, tag: tag))
+            .wrapToIridium()
+            .asRPCRequest()
+        let message = try request.asJSONEncodedString()
+        logger.debug("Publishing payload on topic: \(topic)")
+        try await dispatcher.send(message)
     }
 
     /// Completes with an acknowledgement from the relay network.
-    @discardableResult public func publish(
+    public func publish(
         topic: String,
         payload: String,
         tag: Int,
         prompt: Bool = false,
-        onNetworkAcknowledge: @escaping ((Error?) -> Void)) -> Int64 {
-        let params = RelayJSONRPC.PublishParams(topic: topic, message: payload, ttl: defaultTtl, prompt: prompt, tag: tag)
-        let request = JSONRPCRequest<RelayJSONRPC.PublishParams>(method: RelayJSONRPC.Method.publish.method, params: params)
-        let requestJson = try! request.json()
-        logger.debug("iridium: Publishing Payload on Topic: \(topic)")
+        onNetworkAcknowledge: @escaping ((Error?) -> Void)
+    ) {
+        let rpc = Publish(params: .init(topic: topic, message: payload, ttl: defaultTtl, prompt: prompt, tag: tag))
+        let request = rpc
+            .wrapToIridium()
+            .asRPCRequest()
+        let message = try! request.asJSONEncodedString()
+        logger.debug("Publishing Payload on Topic: \(topic)")
         var cancellable: AnyCancellable?
-        dispatcher.send(requestJson) { [weak self] error in
+        cancellable = requestAcknowledgePublisher
```

Comment:
Need to find way how to move requestAcknowledgePublisher subscription out of this method.

Reply:
Yes, the RelayClient refactor is not over